src/Request.cpp

```diff
@@ -520,6 +526,13 @@ auto Request::getRemainingDownloadBytes() -> ssize_t
     return m_max_download_bytes - m_bytes_written;
 }
 
+auto Request::setHttpStatusCodeFromCurl() -> void
+{
+    long http_response_code = 0;
+    curl_easy_getinfo(m_curl_handle, CURLINFO_RESPONSE_CODE, &http_response_code);
+    m_http_status_code = http::to_enum(static_cast<uint32_t>(http_response_code));
```

Comment:
`uint_t` is better for portability
See: https://confluence.spotxchange.com/pages/viewpage.action?pageId=33623118&focusedCommentId=71467859#comment-71467859

Reply:
I don't think we are concerned about portability at this point and I believe we generally want to be explicit about sizes, so I'm going to leave as is.
If we do decide we want to be portable, though, I would prefer a story to change everything at once instead of being only partially portable.

Reply:
But we do want to write "good" code, especially when open sourcing it. IMO with something like this sooner is better, and starting with this one line won't hurt anything.

Note that `uint_t` is always at least 32-bits

Reply:
On another note, this is a narrowing cast from long. What if curl has a negative or MAX_LONG special code for some reason (who knows)?

Reply:
In my opinion choosing a type that both makes sense and works the same across different platforms _is_ the portable solution. So using explicit sized types is preferred, not just for our code but also for our library code. It's true that this could cause performance issues on 32 bit systems but the complications from supporting different sized types is not worth the minimal benefit of improving performance on a platform most don't care about anymore

Reply:
Good point about the narrowing cast -- looking at that, I noticed the to_enum function could be a little more defensive, so updated that too.

Reply:
The only difference between `uint32_t` and `uint_t` (which is always at least 32 bits) is the former _cannot be larger than 32 bits_, which limits the optimizer.  There's also `uint_least32_t` but that's just overly verbose (and same as `uint`).

What I mean by "portable" is code that can work and be compiled/optimized on different systems (such as windows, embedded, game consoles, unix mainframe, etc).

Reply:
> In my opinion choosing a type that both makes sense and works the same across different platforms is the portable solution.

`uint32_t` and `uint_t` do work the same for this code -- unless you get into what happens with undefined behavior (like left bit-shift).

Reply:
they may work the same in this code, but uint32_t and uint_t do not work the same in all code. Because of this and to maintain a consistent style, we must choose one or the other. The explicitness of the strictly sized types is a great benefit when it comes to portability and though your arguments regarding 32 bit platforms and potential optimizations may be correct, it does not seem to be worth the cost of having different platforms potentially behave differently due to different variable sizes.

Reply:
Fair enough. Thanks for a good discussion!